lib/ood_core/job/adapters/slurm.rb

```diff
@@ -334,13 +334,24 @@ def info(id)
             parse_job_info(v)
           end
 
-          # A job id can return multiple jobs if it corresponds to a job
-          # array id, so we need to find the job that corresponds to the
-          # given job id (if we can't find it, we assume it has completed)
-          info_ary.detect( -> { Info.new(id: id, status: :completed) } ) do |info|
-            # Match the job id or the formatted job & task id "1234_0"
-            info.id == id || info.native[:array_job_task_id] == id
+          # If no job was found we assume that it has completed
+          return Info.new(id: id, status: :completed) unless ! info_ary.empty?
+
+          # If only one job was returned we return it
+          return info_ary.first unless info_ary.length > 1
```

Comment:
See discussion on Torque adapter below. IMHO multiple returns in a method body like this is a bad idea if it can be replaced with a functional equivalent and only one return statement.